### ISSUE TITLE
1.16.5 Dedicated server crash on autocrafting - Exception in server tick loop  java.util.ConcurrentModificationException: null at appeng.util.item.ItemList$ChainedIterator.next

### ISSUE THREAD

**Issue by ProsperCraft** [AppliedEnergistics/Applied-Energistics-2#5158]
**Describe the bug**
A player on our server says he was autocrafting Mekanim things, and the server went down, he was the only player on at the time.

**To Reproduce**
Asking him for steps to reproduce.

**Additional context**
Crashlog-
https://gist.github.com/ProsperCraft/82ef431b0d11ec862898bf98e0bbaeea

**Environment**
appliedenergistics2-8.3.0-alpha.3
forge-1.16.5-36.1.2
Dedicated server.
[TownCraft TWO for 1.16.5 modpack ](https://www.curseforge.com/minecraft/modpacks/towncraft)

**Comment by shartte** [comment]
Do you know what the player was crafting?

**Comment by ProsperCraft** [comment]
Mekanism power cables, but when they did it again, it did not crash.

**Comment by yueh** [comment]
performant might also be a cause here.

**Comment by ProsperCraft** [comment]
It happened again, here is the crash report
https://gist.github.com/ProsperCraft/2bfca0458235159f2e9f4ed2230ce128
Here is the debug log-
https://gist.github.com/ProsperCraft/3478b7575ba6941a58b439533f5a5bd9

**Comment by shartte** [comment]
@ProsperCraft We've just pushed alpha.4 that includes more debugging code to hopefully track the root cause of this down.
In the meantime, I presume the craft that caused this was the same again? Can you describe the pattern that the player is using to craft?

**Comment by zpprsprk** [comment]
I just experienced a similar issue. I was running version alpha.4 and this is the part of the server log file when the crash happened.
https://gist.github.com/zpprsprk/57247faf16fe6affb473826e5d9f4fa3
I was auto crafting with a mekanism ultimate smelting factory. The pattern was a processing pattern for 64 exnihilo nickel ore chunks to 64 nickel ingots. It first happened when I requested 1001 ingots (it said it would do 1024). After restarting the server, I asked for 128 with no issues, then 1024 (exactly) with no issues, then again 1001 (that it said it would do 1024 again) and triggered the crash again.
